SDL2_OpenGL_BoilerPlate/BoilerPlate/Entity.cpp

```diff
@@ -37,17 +49,106 @@ void Entity::Warping(int screenWidth, int screenHeight) {
 
 	//Evaluating x
 	if (position->x < -width) {
+
 		position->x = width;
 	}
 	else if (position->x > width) {
+
 		position->x = -width;
 	}
 
 	//Evaluationg y
 	if (position->y < -height) {
+
 		position->y = height;
 	}
 	else if (position->y > height) {
+
 		position->y = -height;
 	}
 }
+
+float Entity::CalculateRadius() {
+	
+	float farthestAxis = 0;
+
+	for (int i = 0; i < entityVertexContainer.size(); i++) {
+
+		if (fabsf(entityVertexContainer[i].x) > farthestAxis) {
+		
+			farthestAxis = fabsf(entityVertexContainer[i].x);
+		}
+		
+		if (fabsf(entityVertexContainer[i].y) > farthestAxis) {
+
+			farthestAxis = fabsf(entityVertexContainer[i].y);
+		}
+	}
+
+	return farthestAxis;
+}
+
+void Entity::EntityDebugger() {
+
+	if (debuggerState == true) {
+
+		DrawHitCircle();
+	}
+}
+
+void Entity::DrawHitCircle() {
+
+	MathUtilities mathUtilities;
+
+	int lineAmount = 100;
+
+	float twicePi = 2.0f * mathUtilities.PI;
+
+	glLoadIdentity();
```

Comment:
Rather than a hit circle, this is just a circle. Remember to describe why you reset the transform matrix to identity.